Sources/JSONRequest.swift

```diff
@@ -150,19 +150,18 @@ open class JSONRequest {
                            headers: JSONObject? = nil,
                            timeOut: TimeInterval? = nil) -> JSONResult {
 
-        let semaphore = DispatchSemaphore(value: 0)
         var requestResult: JSONResult = JSONResult.failure(error: JSONError.unknownError,
                                                            response: nil, body: nil)
+
+        let group = DispatchGroup()
+        group.enter()
         submitAsyncRequest(method: method, url: url, queryParams: queryParams,
                            payload: payload, headers: headers, timeOut: timeOut) { result in
                             requestResult = result
-                            semaphore.signal()
+                            group.leave()
         }
         // Wait for the request to complete
-        while semaphore.wait(timeout: DispatchTime.now()) == .timedOut {
```

Comment:
what if replace this polling just with `semaphore.wait()`?

Reply:
Yeah I thought about that too... I think I meant to try it and then it slipped my mind. I'll try it now to see what results I get.

Reply:
You're right... updated to use a simple semaphore.